src/core/llm/gemini.ts

```
@@ -369,24 +369,61 @@ export class GeminiProvider extends BaseLLMProvider<
+      // 1. Ensure the parameter type is STRING.
+      //    (Verify 'SchemaType.STRING' is correct from your '@google/generative-ai' imports; it might be "STRING").
+      // 2. Ensure the 'format' field actually exists on the schema.
+      if (newParamSchema.type === SchemaType.STRING && 
+          Object.prototype.hasOwnProperty.call(newParamSchema, 'format') &&
+          newParamSchema.format !== undefined && newParamSchema.format !== null) {
+        
+        const currentFormat = String(newParamSchema.format).toLowerCase();
+
+        if (currentFormat !== 'enum' && currentFormat !== 'date-time') {
+          // If format exists and is NOT 'enum' or 'date-time', delete it.
+          delete newParamSchema.format;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)